### PR TITLE
Move to `.NET 5 preview.1`

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
# PR Summary

Updated associated build files to netcoreapp5.0 and ref assemblies to 5.0-preview.1 where applicable.  Updated files.wxs to changes.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # Licensed under the MIT License. See https://go.microsoft.com/fwlink/?linkid=2090316 for license information.
```
Comment (PoshChan) [inline .devcontainer/Dockerfile]: @SteveL-MSFT, your last commit had 8 failures in `PowerShell-CI-linux`
(These are 5 of the failures)

NullConditionalMemberAccess.?. operator tests.Can get member value of a non-null variable
```powershell
Expected like wildcard 'pwsh*' to match 'ConsoleHost mai', but it did not match.
at <ScriptBlock>, : line 5
5:             (Get-Process -Id $PID)?.Name | Should -BeLike "pwsh*"
```
NullConditionalMemberAccess.?. operator tests.Use ?. on a dynamic property name
```powershell
Expected like wildcard 'pwsh*' to match 'ConsoleHost mai', but it did not match.
at <ScriptBlock>, : line 2
2:             (Get-Process -Id $PID)?.$propName | Should -BeLike 'pwsh*'
```
Validate start of console host.No new assemblies are loaded
```powershell
Expected exactly $null, but got @(@{InputObject=System.Buffers.dll; SideIndicator=<=}, @{InputObject=System.Diagnostics.Debug.dll; SideIndicator=<=}, @{InputObject=System.Resources.ResourceManager.dll; SideIndicator=<=}, @{InputObject=System.Runtime.Extensions.dll; SideIndicator=<=}, @{InputObject=System.Threading.Tasks.dll; SideIndicator=<=}).
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Host/Startup.Tests.ps1: line 125
125:         $diffs | Should -BeExactly $null
```
NullConditionalMemberAccess.?. operator tests.Can get member value of a non-null variable
```powershell
Expected like wildcard 'pwsh*' to match 'ConsoleHost mai', but it did not match.
at <ScriptBlock>, : line 5
5:             (Get-Process -Id $PID)?.Name | Should -BeLike "pwsh*"
```
NullConditionalMemberAccess.?. operator tests.Use ?. on a dynamic property name
```powershell
Expected like wildcard 'pwsh*' to match 'ConsoleHost mai', but it did not match.
at <ScriptBlock>, : line 2
2:             (Get-Process -Id $PID)?.$propName | Should -BeLike 'pwsh*'
```

Comment (PoshChan) [inline .devcontainer/Dockerfile]: @SteveL-MSFT, your last commit had 2 failures in `PowerShell-CI-macos`
Validate start of console host.No new assemblies are loaded
```powershell
Expected exactly $null, but got @(@{InputObject=System.Buffers.dll; SideIndicator=<=}, @{InputObject=System.Diagnostics.Debug.dll; SideIndicator=<=}, @{InputObject=System.Resources.ResourceManager.dll; SideIndicator=<=}, @{InputObject=System.Runtime.Extensions.dll; SideIndicator=<=}, @{InputObject=System.Threading.Tasks.dll; SideIndicator=<=}).
at <ScriptBlock>, /Users/runner/runners/2.165.0/work/1/s/test/powershell/Host/Startup.Tests.ps1: line 125
125:         $diffs | Should -BeExactly $null
```
Invoke-WebRequest tests.Denial of service.Image Parsing
```powershell
Expected the actual value to be greater than 5, but got 0.316031693749085.
at <ScriptBlock>, /Users/runner/runners/2.165.0/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1: line 1932
1932:             $pathologicalRatio | Should -BeGreaterThan 5
```

Comment (PoshChan) [inline .devcontainer/Dockerfile]: @SteveL-MSFT, your last commit had 1 failures in `PowerShell-CI-windows`
Invoke-WebRequest tests.Denial of service.Image Parsing
```powershell
Expected the actual value to be greater than 5, but got 0.29322238885395.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\WebCmdlets.Tests.ps1: line 1932
1932:             $pathologicalRatio | Should -BeGreaterThan 5
```

Comment (PoshChan) [inline .devcontainer/Dockerfile]: @SteveL-MSFT, your last commit had 8 failures in `PowerShell-CI-linux`
(These are 5 of the failures)

NullConditionalMemberAccess.?. operator tests.Can get member value of a non-null variable
```powershell
Expected like wildcard 'pwsh*' to match 'ConsoleHost mai', but it did not match.
at <ScriptBlock>, : line 5
5:             (Get-Process -Id $PID)?.Name | Should -BeLike "pwsh*"
```
NullConditionalMemberAccess.?. operator tests.Use ?. on a dynamic property name
```powershell
Expected like wildcard 'pwsh*' to match 'ConsoleHost mai', but it did not match.
at <ScriptBlock>, : line 2
2:             (Get-Process -Id $PID)?.$propName | Should -BeLike 'pwsh*'
```
Validate start of console host.No new assemblies are loaded
```powershell
Expected exactly $null, but got @(@{InputObject=System.Diagnostics.DiagnosticSource.dll; SideIndicator=<=}, @{InputObject=System.Net.NameResolution.dll; SideIndicator=<=}, @{InputObject=System.Private.Xml.Linq.dll; SideIndicator=<=}, @{InputObject=System.Xml.XDocument.dll; SideIndicator=<=}).
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Host/Startup.Tests.ps1: line 124
124:         $diffs | Should -BeExactly $null
```
NullConditionalMemberAccess.?. operator tests.Can get member value of a non-null variable
```powershell
Expected like wildcard 'pwsh*' to match 'ConsoleHost mai', but it did not match.
at <ScriptBlock>, : line 5
5:             (Get-Process -Id $PID)?.Name | Should -BeLike "pwsh*"
```
NullConditionalMemberAccess.?. operator tests.Use ?. on a dynamic property name
```powershell
Expected like wildcard 'pwsh*' to match 'ConsoleHost mai', but it did not match.
at <ScriptBlock>, : line 2
2:             (Get-Process -Id $PID)?.$propName | Should -BeLike 'pwsh*'
```

Comment (PoshChan) [inline .devcontainer/Dockerfile]: @SteveL-MSFT, your last commit had 3 failures in `PowerShell-CI-macos`
Validate start of console host.No new assemblies are loaded
```powershell
Expected exactly $null, but got @(@{InputObject=System.Diagnostics.DiagnosticSource.dll; SideIndicator=<=}, @{InputObject=System.Net.NameResolution.dll; SideIndicator=<=}, @{InputObject=System.Private.Xml.Linq.dll; SideIndicator=<=}, @{InputObject=System.Xml.XDocument.dll; SideIndicator=<=}).
at <ScriptBlock>, /Users/runner/runners/2.165.0/work/1/s/test/powershell/Host/Startup.Tests.ps1: line 124
124:         $diffs | Should -BeExactly $null
```
Invoke-WebRequest tests.Denial of service.Image Parsing
```powershell
Expected the actual value to be greater than 5, but got 0.318270278323541.
at <ScriptBlock>, /Users/runner/runners/2.165.0/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1: line 1932
1932:             $pathologicalRatio | Should -BeGreaterThan 5
```
Test-Connection.TraceRoute.TraceRoute works
```powershell
Expected exactly '10.79.1.148', but got 127.0.0.1.
at <ScriptBlock>, /Users/runner/runners/2.165.0/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1: line 257
257:             $result[0].TargetAddress | Should -BeExactly $realAddress
```

Comment (SteveL-MSFT) [reply .devcontainer/Dockerfile]: The nullconditional tests are failing due to regression in .NET where [Process.ProcessName not returning pwsh]()https://github.com/dotnet/runtime/issues/33673) on Linux

Comment (PoshChan) [inline .devcontainer/Dockerfile]: @SteveL-MSFT, your last commit had 1 failures in `PowerShell-CI-windows`
Invoke-WebRequest tests.Denial of service.Image Parsing
```powershell
Expected the actual value to be greater than 5, but got 0.296258838270831.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\WebCmdlets.Tests.ps1: line 1932
1932:             $pathologicalRatio | Should -BeGreaterThan 5
```

Comment (PoshChan) [inline .devcontainer/Dockerfile]: @SteveL-MSFT, your last commit had 4 failures in `PowerShell-CI-linux`
Validate start of console host.No new assemblies are loaded
```powershell
Expected exactly $null, but got @(@{InputObject=System.Diagnostics.DiagnosticSource.dll; SideIndicator=<=}, @{InputObject=System.Net.NameResolution.dll; SideIndicator=<=}, @{InputObject=System.Private.Xml.Linq.dll; SideIndicator=<=}, @{InputObject=System.Xml.XDocument.dll; SideIndicator=<=}).
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Host/Startup.Tests.ps1: line 124
124:         $diffs | Should -BeExactly $null
```
Using of ternary operator.Ternary expression '{ $($p = Get-Process -Id $PID; $p.Name -eq 'pwsh') ? 'Core' : 'Desktop' }' should generate correct results
```powershell
Expected strings to be the same, but they were different.
Expected length: 4
Actual length:   7
Strings differ at index 0.
Expected: 'Core'
But was:  'Desktop'
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Language/Operators/TernaryOperator.Tests.ps1: line 56
56:         & $Script | Should -BeExactly $ExpectedValue
```
Using of ternary operator.Ternary expression '{ $IsCoreCLR -and (Get-Process -Id $PID).Name -eq 'pwsh' ? 'Core' : 'Desktop' }' should generate correct results
```powershell
Expected strings to be the same, but they were different.
Expected length: 4
Actual length:   7
Strings differ at index 0.
Expected: 'Core'
But was:  'Desktop'
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Language/Operators/TernaryOperator.Tests.ps1: line 56
56:         & $Script | Should -BeExactly $ExpectedValue
```
Invoke-WebRequest tests.Denial of service.Image Parsing
```powershell
Expected the actual value to be greater than 5, but got 0.269696349876346.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1: line 1932
1932:             $pathologicalRatio | Should -BeGreaterThan 5
```

Comment (PoshChan) [inline .devcontainer/Dockerfile]: @SteveL-MSFT, your last commit had 3 failures in `PowerShell-CI-macos`
Validate start of console host.No new assemblies are loaded
```powershell
Expected exactly $null, but got @(@{InputObject=System.Diagnostics.DiagnosticSource.dll; SideIndicator=<=}, @{InputObject=System.Net.NameResolution.dll; SideIndicator=<=}, @{InputObject=System.Private.Xml.Linq.dll; SideIndicator=<=}, @{InputObject=System.Xml.XDocument.dll; SideIndicator=<=}).
at <ScriptBlock>, /Users/runner/runners/2.165.0/work/1/s/test/powershell/Host/Startup.Tests.ps1: line 124
124:         $diffs | Should -BeExactly $null
```
Invoke-WebRequest tests.Denial of service.Image Parsing
```powershell
Expected the actual value to be greater than 5, but got 0.330475306456397.
at <ScriptBlock>, /Users/runner/runners/2.165.0/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1: line 1932
1932:             $pathologicalRatio | Should -BeGreaterThan 5
```
Test-Connection.TraceRoute.TraceRoute works
```powershell
Expected exactly '127.0.0.1', but got 10.79.10.245.
at <ScriptBlock>, /Users/runner/runners/2.165.0/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1: line 257
257:             $result[0].TargetAddress | Should -BeExactly $realAddress
```

Comment (PoshChan) [inline .devcontainer/Dockerfile]: @SteveL-MSFT, your last commit had 1 failures in `PowerShell-CI-windows`
Invoke-WebRequest tests.Denial of service.Image Parsing
```powershell
Expected the actual value to be greater than 5, but got 0.263530095582537.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\WebCmdlets.Tests.ps1: line 1932
1932:             $pathologicalRatio | Should -BeGreaterThan 5
```

**Comment by SteveL-MSFT** [comment]
It looks like there's an issue in .NET 5 preview1 where we set the main thread name and that somehow gets propagated as the process name on Linux (only) and so tests that check if it's pwsh ends up failing.  I'll see about putting in some reasonable workarounds for now and add a pending test to verify the process name.

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (PoshChan) [inline .devcontainer/Dockerfile]: @SteveL-MSFT, your last commit had 2 failures in `PowerShell-CI-linux`
Validate start of console host.No new assemblies are loaded
```powershell
Expected exactly $null, but got @(@{InputObject=System.Diagnostics.DiagnosticSource.dll; SideIndicator=<=}, @{InputObject=System.Net.NameResolution.dll; SideIndicator=<=}, @{InputObject=System.Private.Xml.Linq.dll; SideIndicator=<=}, @{InputObject=System.Xml.XDocument.dll; SideIndicator=<=}).
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Host/Startup.Tests.ps1: line 124
124:         $diffs | Should -BeExactly $null
```
Invoke-WebRequest tests.Denial of service.Image Parsing
```powershell
Expected the actual value to be greater than 5, but got 0.248258769166142.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1: line 1932
1932:             $pathologicalRatio | Should -BeGreaterThan 5
```

Comment (PoshChan) [inline .devcontainer/Dockerfile]: @SteveL-MSFT, your last commit had 2 failures in `PowerShell-CI-macos`
Validate start of console host.No new assemblies are loaded
```powershell
Expected exactly $null, but got @(@{InputObject=System.Diagnostics.DiagnosticSource.dll; SideIndicator=<=}, @{InputObject=System.Net.NameResolution.dll; SideIndicator=<=}, @{InputObject=System.Private.Xml.Linq.dll; SideIndicator=<=}, @{InputObject=System.Xml.XDocument.dll; SideIndicator=<=}).
at <ScriptBlock>, /Users/runner/runners/2.165.2/work/1/s/test/powershell/Host/Startup.Tests.ps1: line 124
124:         $diffs | Should -BeExactly $null
```
Invoke-WebRequest tests.Denial of service.Image Parsing
```powershell
Expected the actual value to be greater than 5, but got 0.22236631422938.
at <ScriptBlock>, /Users/runner/runners/2.165.2/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1: line 1932
1932:             $pathologicalRatio | Should -BeGreaterThan 5
```

**Review by TravisEz13 — COMMENTED** [review]
I've looked at the changes so far

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (PoshChan) [inline .devcontainer/Dockerfile]: @SteveL-MSFT, your last commit had 1 failures in `PowerShell-CI-linux`
Invoke-WebRequest tests.Denial of service.Image Parsing
```powershell
Expected the actual value to be greater than 5, but got 0.273896892271451.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1: line 1932
1932:             $pathologicalRatio | Should -BeGreaterThan 5
```

Comment (PoshChan) [inline .devcontainer/Dockerfile]: @SteveL-MSFT, your last commit had 1 failures in `PowerShell-CI-macos`
Invoke-WebRequest tests.Denial of service.Image Parsing
```powershell
Expected the actual value to be greater than 5, but got 0.989304608202352.
at <ScriptBlock>, /Users/runner/runners/2.165.2/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1: line 1932
1932:             $pathologicalRatio | Should -BeGreaterThan 5
```

**Comment by SteveL-MSFT** [comment]
.NET 5 specifically has improvements to regex, so needed to increase the size of the DOS test

**Comment by SteveL-MSFT** [comment]
On Windows, pwsh is crashing in a native API call.

**Comment by danmosemsft** [comment]
> .NET 5 specifically has improvements to regex, so needed to increase the size of the DOS test

cc @stephentoub  👏

**Comment by danmosemsft** [comment]
> It looks like there's an issue in .NET 5 preview1 where we set the main thread name and that somehow gets propagated as the process name on Linux (only) 

@SteveL-MSFT is that a bug in .NET?

**Comment by SteveL-MSFT** [comment]
@danmosemsft The main thread name becoming the process name is a bug in .NET and is understood (https://github.com/dotnet/runtime/issues/33673).  That issue isn't blocking as it's currently only affecting tests.  The crashing issue (https://github.com/dotnet/runtime/issues/33703) is preventing us from moving to .NET 5.  Thanks!

**Comment by TravisEz13** [comment]
There is already an issues filed for regex DOS tests:  https://github.com/PowerShell/PowerShell/issues/12138

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (PoshChan) [inline .devcontainer/Dockerfile]: @SteveL-MSFT, your last commit had 1 failures in `PowerShell-CI-linux`
Invoke-WebRequest tests.Denial of service.Image Parsing
```powershell
Expected the actual value to be greater than 5, but got 3.69898655485732.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1: line 1932
1932:             $pathologicalRatio | Should -BeGreaterThan 5
```

Comment (PoshChan) [inline .devcontainer/Dockerfile]: @SteveL-MSFT, your last commit had 1 failures in `PowerShell-CI-windows`
Validate start of console host.No new assemblies are loaded
```powershell
Expected exactly $null, but got @{InputObject=System.Security.Principal.dll; SideIndicator=<=}.
at <ScriptBlock>, D:\a\1\s\test\powershell\Host\Startup.Tests.ps1: line 120
120:         $diffs | Should -BeExactly $null
```

Comment (PoshChan) [inline .devcontainer/Dockerfile]: @SteveL-MSFT, your last commit had 1 failures in `PowerShell-CI-macos`
Test-Connection.Ping.Force IPv4 with implicit PingOptions
```powershell
Expected exactly '127.0.0.1', but got 10.79.1.122.
at <ScriptBlock>, /Users/runner/runners/2.165.2/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1: line 83
83:             $result[0].Address | Should -BeExactly $realAddress
```

**Comment by SteveL-MSFT** [comment]
@PoshChan retry macos

**Comment by PoshChan** [comment]
@SteveL-MSFT, successfully started retry of `PowerShell-CI-macOS`

File: test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1 [hunk 2]
```diff
@@ -1904,7 +1904,7 @@ Describe "Invoke-WebRequest tests" -Tags "Feature", "RequireAdminOnWindows" {
         It "Image Parsing" {
             $dosUri = Get-WebListenerUrl -Test 'Dos' -query @{
                 dosType='img'
-                dosLength='5000'
```
Comment (TravisEz13) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1]: Discuss offline

**Review by TravisEz13 — CHANGES_REQUESTED** [review]

File: test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1 (hunk 2 above)
Comment (TravisEz13) [reply test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1]: Delete the test, the old regex is no longer pathological due to changes in .NET.  Verified with .NET that they did optimizations.

**Review by TravisEz13 — APPROVED** [review]

**Comment by TravisEz13** [comment]
.NET originally said that the perf improvement was due to optimization, but it's actually due to a bug.  I filed an issue to add it back to avoid delaying this PR.
https://github.com/PowerShell/PowerShell/issues/12155

**Comment by stephentoub** [comment]
> .NET originally said that the perf improvement was due to optimization, but it's actually due to a bug

It's half and half :)  The algorithmic complexity improved, in part due to perf improvements, and in part due to a bug fixed in Preview 2 (https://github.com/dotnet/runtime/pull/33409).

**Pull request merged**